### PR TITLE
Adds CBMC proof for aws_cryptosdk_default_cmm_set_alg_id function

### PR DESCRIPTION
**Pull request by tegansb**
*Description of changes:*

Adds harness, Makefile and yaml for aws_cryptosdk_default_cmm_set_alg_id. 
Adds preconditions to function. 
Updates ensure_cryptosdk_keyring_has_allocated_members. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by alex-chew — APPROVED** [review]
LGTM

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+void aws_cryptosdk_default_cmm_set_alg_id_harness() {
+    /* Nondet input */
+    enum aws_cryptosdk_alg_id alg_id;
+
+    const struct aws_cryptosdk_keyring_vt vtable = { .vt_size    = sizeof(struct aws_cryptosdk_keyring_vt),
+                                                     .name       = ensure_c_str_is_allocated(SIZE_MAX),
+                                                     .destroy    = nondet_voidp(),
+                                                     .on_encrypt = nondet_voidp(),
+                                                     .on_decrypt = nondet_voidp() };
+    struct aws_cryptosdk_keyring *keyring        = malloc(sizeof(*keyring));
```
Comment (adpaco-aws) [inline verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:32]: Can we put all these declarations together?
```suggestion
    enum aws_cryptosdk_alg_id alg_id;
    struct aws_cryptosdk_keyring *keyring        = malloc(sizeof(*keyring));
    const struct aws_cryptosdk_keyring_vt vtable = { .vt_size    = sizeof(struct aws_cryptosdk_keyring_vt),
                                                     .name       = ensure_c_str_is_allocated(SIZE_MAX),
                                                     .destroy    = nondet_voidp(),
                                                     .on_encrypt = nondet_voidp(),
                                                     .on_decrypt = nondet_voidp() };

```

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ensure_cryptosdk_keyring_has_allocated_members(keyring, &vtable);
+    __CPROVER_assume(aws_cryptosdk_keyring_is_valid(keyring));
+
+    /* Instantiate the default (non-caching) implementation of the Crypto MaterialsManager (CMM).*/
+    struct aws_cryptosdk_cmm *cmm = aws_cryptosdk_default_cmm_new(can_fail_allocator(), keyring);
+
+    /*Assumptions */
```
Comment (adpaco-aws) [inline verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:39]:
```suggestion
    /* Assumptions */
```

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const struct aws_cryptosdk_keyring_vt vtable = { .vt_size    = sizeof(struct aws_cryptosdk_keyring_vt),
+                                                     .name       = ensure_c_str_is_allocated(SIZE_MAX),
+                                                     .destroy    = nondet_voidp(),
+                                                     .on_encrypt = nondet_voidp(),
+                                                     .on_decrypt = nondet_voidp() };
+    struct aws_cryptosdk_keyring *keyring        = malloc(sizeof(*keyring));
+    ensure_cryptosdk_keyring_has_allocated_members(keyring, &vtable);
+    __CPROVER_assume(aws_cryptosdk_keyring_is_valid(keyring));
+
+    /* Instantiate the default (non-caching) implementation of the Crypto MaterialsManager (CMM).*/
```
Comment (adpaco-aws) [inline verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:36]:
```suggestion
    /* Instantiate the default (non-caching) implementation of the Crypto MaterialsManager (CMM) */
```

**Review by adpaco-aws — CHANGES_REQUESTED** [review]
Aside from the minor changes that I suggested, I think this branch needs to be rebased against master.

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c (hunk 1 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:32]: Sure! I've updated this.

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c (hunk 2 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:39]: Updated!

File: verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c (hunk 3 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_default_cmm_set_alg_id/aws_cryptosdk_default_cmm_set_alg_id_harness.c:36]: Updated!

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**